Content.Server/Body/Respiratory/RespiratorComponent.cs

```diff
@@ -358,27 +347,19 @@ private void TakeSuffocationDamage()
                 alertsComponent.ShowAlert(AlertType.LowOxygen);
             }
 
-            if (!Owner.TryGetComponent(out IDamageableComponent? damageable))
-            {
-                return;
-            }
-
-            damageable.TryChangeDamage(DamageType, _damage, false);
+            Owner.EntityManager.EventBus.RaiseLocalEvent(Owner.Uid, new TryChangeDamageEvent(Damage, true), false);
```

Comment:
In general I've removed 'TryGetComponent' or 'HasComponent' tests in favour of just raising TryChangeDamage events (which just do nothing if there is no component). But maybe events are computationally expensive and checking for applicable components beforehand is proffered?

Reply:
they're not much different perf wise, but should get another maintainers opinion as to whether this should be a damageable system public method or an event; i'm leaning towards a public method, since its unlikely anything else will need to subscribe to this event (damageable can raise its own event for receiving feedback)